app/vminsert/netstorage/insert_ctx.go

```diff
@@ -21,8 +22,9 @@ type InsertCtx struct {
 	Labels        []prompb.Label
 	MetricNameBuf []byte
 
-	bufRowss  []bufRows
-	labelsBuf []byte
+	groupBufRowss           map[string][]bufRows
+	labelsBuf               []byte
+	failedReplicationGroups map[string]bool
```

Comment:
This field isn't needed if errors are counted in the loop over `replicationGroups` as described above.